test/Fencer/Server/Test.hs

```diff
@@ -68,6 +77,42 @@ test_serverResponseNoRules =
       , Proto.rateLimitRequestHitsAddend = 0
       }
 
+-- | Test that when Fencer is started and there is no configuration
+-- directory, requests to Fencer will be responded to with OK.
```

Comment:
The tests ensures that a single request is responded with OK. Should we test a sequence of requests? Are there different types of requests? If so, it would be nice to test those too.

Sorry, I should invest some time in studying how `lyft/ratelimit` works.

Reply:
There is only one request type:

https://github.com/juspay/fencer/blob/97e62225a6e7d0fbdf9e3ab554e3e44ac427e398/lib/Fencer/Proto.hs#L85-L90

Submitting a request in this situation does not change the server's state, so having a sequence would make no difference.

Reply:
> Submitting a request in this situation does not change the server's state, so having a sequence would make no difference.

The point of tests is not to rely on any intended behavior, but to showcase and persist this behavior.

And how do I know that the server doesn't change its state or won't accidentally throw an exception internally after returning `OK` on first request? Someone (looks at self) might change the code later and break this invariant without anyone noticing.

So I think that if the invariant is that the server responds with OK to several requests in a row, then we should test that,

Reply:
I changed it to multiple requests now, confirming the invariant that the server state doesn't change for such situations.

Reply:
Thanks!